File: backend/src/main/java/com/actividades/backend/model/Evento.java
package com.actividades.backend.model;

import jakarta.persistence.*;

@Entity
public class Evento {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false)
    private String titulo;
    @Column(nullable = false)
    private String ciudad;
    @Column(nullable = false)
    private String categoria;

    public Evento() {
    }

    public Evento(String titulo, String ciudad, String categoria) {
        this.titulo = titulo;
        this.ciudad = ciudad;
        this.categoria = categoria;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getCategoria() {
        return categoria;
    }

    public void setCategoria(String categoria) {
        this.categoria = categoria;
    }

    public String getCiudad() {
        return ciudad;
    }

    public void setCiudad(String ciudad) {
        this.ciudad = ciudad;
    }

    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

}

File: evento_app/lib/models/evento.dart
class Evento {
  final int id;
  final String titulo;
  final String ciudad;
  final String categoria;

  Evento({
    required this.id,
    required this.titulo,
    required this.ciudad,
    required this.categoria,
  });

  factory Evento.fromJson(Map<String, dynamic> json) {
    return Evento(
      id: json['id'],
      titulo: json['titulo'],
      ciudad: json['ciudad'],
      categoria: json['categoria'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'titulo': titulo,
      'ciudad': ciudad,
      'categoria': categoria,
    };
  }
}